components/brave_shields/browser/custom_subscription_download_manager.h

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_CUSTOM_SUBSCRIPTION_DOWNLOAD_MANAGER_H_
+#define BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_CUSTOM_SUBSCRIPTION_DOWNLOAD_MANAGER_H_
+
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+
+#include "base/memory/weak_ptr.h"
+#include "components/download/public/background_service/download_params.h"
+
+#include "components/services/storage/public/mojom/blob_storage_context.mojom.h"
+
+namespace storage {
+class BlobDataHandle;
+}
+
+namespace download {
+class DownloadService;
+}  // namespace download
+
+namespace brave_shields {
+
+class AdBlockSubscriptionServiceManager;
+
+class CustomSubscriptionDownloadClient;
+
+namespace proto {
+class PredictionModel;
+}  // namespace proto
+
+// Manages the downloads of prediction models.
+class CustomSubscriptionDownloadManager {
+ public:
+  CustomSubscriptionDownloadManager(
+      download::DownloadService* download_service,
+      AdBlockSubscriptionServiceManager* subscription_manager,
+      scoped_refptr<base::SequencedTaskRunner> background_task_runner);
+  virtual ~CustomSubscriptionDownloadManager();
+  CustomSubscriptionDownloadManager(const CustomSubscriptionDownloadManager&) =
+      delete;
+  CustomSubscriptionDownloadManager& operator=(
+      const CustomSubscriptionDownloadManager&) = delete;
+
+  // Starts a download for |download_url|. Will schedule a higher priority
+  // download if |from_ui| is true.
+  virtual void StartDownload(const GURL& download_url, bool from_ui);
+
+  // Cancels all pending downloads.
+  virtual void CancelAllPendingDownloads();
+
+  // Returns whether the downloader can download models.
```

Comment:
there are no models anymore. I guess when we remove all unused methods then the whole thing would look less scary

Reply:
Actually, now that I look at it again we probably do need to check `IsAvailableForDownloads` before we try to start any downloads. I updated the comment though.